internal/dag/gatewayapi_processor.go

```diff
@@ -106,11 +136,44 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 
 		// Process all the routes that match this Gateway.
 		for _, matchingRoute := range matchingRoutes {
-			p.computeHTTPRoute(matchingRoute)
+			hosts := p.computeHosts(matchingRoute)
+			p.computeHTTPRoute(matchingRoute, hosts, listenerSecret)
 		}
 	}
 }
 
+func (p *GatewayAPIProcessor) computeHosts(route *gatewayapi_v1alpha1.HTTPRoute) []string {
```

Comment:
I expect https://github.com/kubernetes-sigs/gateway-api/pull/597 to merge upstream. I suggest aligning as much as possible to ease the transition to upstream.

Reply:
Yeah for sure, when that merges we can update and remove this code if it's not required any longer.